### PR TITLE
Add Cosmote TV

### PR DESCRIPTION
**Pull request by down-bad**
Cosmote TV is a Greek pay television service. This presence should support all functionality of the website, such as watching live TV, replay TV, streaming movies, series, and all the pages of the site.

Home page
Screenshot 2021-09-11 022045
Live TV
Screenshot 2021-09-11 022340
Replay TV
Screenshot 2021-09-11 022501
Movie/Series/etc
Screenshot 2021-09-11 022747

### CODE REVIEW

**Comment by down-bad** [comment]
Changes image based on the channel (supports all available channels)

Examples:
Screenshot 2021-09-11 144757
Screenshot 2021-09-11 144614

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:102]:
```suggestion
      const series = document.querySelector(
```

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:85]:
```suggestion
     timestamps = presence.getTimestamps(Math.floor(currentTime), Math.floor(duration)),
```

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:84]:
```suggestion
     { paused, currentTime, duration } = video,
```

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:18]:
```suggestion
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:11]:
```suggestion
```

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:51]:
```suggestion
```

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:14]:
```suggestion
    largeImageKey: "tv",
    startTimestamp = Math.floor(Date.now() / 1000)
```

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:65]:
```suggestion
```

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:67]:
```suggestion
```

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:77]:
```suggestion
    if (!live) {
```

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:70]:
```suggestion
```

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:116]:
```suggestion
    if (channel.length === 0) {
```

File: websites/C/Cosmote TV/presence.ts
Comment (EncryptedDev) [inline websites/C/Cosmote TV/presence.ts:117]:
```suggestion
      if (!live)
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Comment by EncryptedDev** [comment]
Upload further proof of working please

**Comment by down-bad** [comment]
Home
Screenshot 2021-09-12 122105
Screenshot 2021-09-12 122132

Channels
Screenshot 2021-09-12 122154
Screenshot 2021-09-12 123623

Series
Screenshot 2021-09-12 123206
Screenshot 2021-09-12 123219
etc...

Live TV
Screenshot 2021-09-12 122344
Screenshot 2021-09-12 123558

Replay TV
Screenshot 2021-09-12 122523
Screenshot 2021-09-12 122538

Movie/Series/other
Screenshot 2021-09-12 123117
Screenshot 2021-09-12 123043

Paused
Screenshot 2021-09-12 123954
Screenshot 2021-09-12 124009

**Review by EncryptedDev — DISMISSED** [review]
Seems cool with me, await proper approval from maintainers shortly.

File: websites/C/Cosmote TV/presence.ts
Comment (Slowlife01) [inline websites/C/Cosmote TV/presence.ts:19]:
```suggestion
    data.smallImageText = (await strings).search;
```

File: websites/C/Cosmote TV/presence.ts
Comment (Slowlife01) [inline websites/C/Cosmote TV/presence.ts:15]:
```suggestion
  }),
  startedTime = Math.floor(Date.now() / 1000);

presence.on("UpdateData", async () => {
  const data: PresenceData = {
    largeImageKey: "tv",
    startTimestamp: startedTime
  };
```

File: websites/C/Cosmote TV/presence.ts
Comment (Slowlife01) [inline websites/C/Cosmote TV/presence.ts:76]:
```suggestion
      data.smallImageKey = paused ? "pause" : "play";
      data.smallImageText = paused
          ? (await strings).pause
          : (await strings).play;
```

File: websites/C/Cosmote TV/presence.ts
Comment (Slowlife01) [inline websites/C/Cosmote TV/presence.ts:79]:
```suggestion
      data.endTimestamp = timestamps.pop();
```

File: websites/C/Cosmote TV/presence.ts
Comment (Slowlife01) [inline websites/C/Cosmote TV/presence.ts:98]:
```suggestion
      data.smallImageKey = paused ? "pause" : "live";
      data.smallImageText = paused
          ? (await strings).pause
          : (await strings).live;
```

File: websites/C/Cosmote TV/presence.ts
Comment (Slowlife01) [inline websites/C/Cosmote TV/presence.ts:111]:
```suggestion
    if (channel?.length !== 0) {
```

**Review by Slowlife01 — CHANGES_REQUESTED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**
